### ISSUE TITLE
Losing glibc dependency

### ISSUE THREAD

**Issue by am11** [yuk7/AlpineWSL#7]
One of the main attraction of Alpine Linux distribution is that it uses muscle libc instead of GNU one. The former is much lighter and anything that links with this libc on the system gets smaller code size. Most of the mainstream software packages are available via Alpine's aports. The hard requirement of glibc is fading away.

Could we have a version of AlpineWSL in this repo that doesn't have glibc?

Proposal:

```diff
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,6 @@ LNCR_EXE=Alpine.exe
 DLR=curl
 DLR_FLAGS=-L
 BASE_URL=http://dl-cdn.alpinelinux.org/alpine/v3.8/releases/x86_64/alpine-minirootfs-3.8.0-x86_64.tar.gz
-GLIBC_URL=https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.28-r0/glibc-2.28-r0.apk
 LNCR_ZIP_URL=https://github.com/yuk7/wsldl/releases/download/18102701/icons.zip
 LNCR_ZIP_EXE=Alpine.exe

@@ -36,14 +35,11 @@ rootfs.tar.gz: rootfs
        cd rootfs; sudo tar -zcpf ../rootfs.tar.gz `sudo ls`
        sudo chown `id -un` rootfs.tar.gz

-rootfs: base.tar.gz glibc.apk
+rootfs: base.tar.gz
        @echo -e '\e[1;31mBuilding rootfs...\e[m'
        mkdir rootfs
        sudo tar -zxpf base.tar.gz -C rootfs
        sudo cp -f /etc/resolv.conf rootfs/etc/resolv.conf
-       sudo cp -f glibc.apk rootfs/
-       sudo chroot rootfs /sbin/apk add /glibc.apk --allow-untrusted
-       sudo rm -rf rootfs/glibc.apk
        sudo chroot rootfs /sbin/apk update
        sudo chroot rootfs /sbin/apk add bash
        echo "# This file was automatically generated by WSL. To stop automatic generation of this file, remove this line." | sudo tee rootfs/etc/resolv.conf
@@ -54,10 +50,6 @@ base.tar.gz:
        @echo -e '\e[1;31mDownloading base.tar.gz...\e[m'
        $(DLR) $(DLR_FLAGS) $(BASE_URL) -o base.tar.gz

-glibc.apk:
-       @echo -e '\e[1;31mDownloading glibc.apk...\e[m'
-       $(DLR) $(DLR_FLAGS) $(GLIBC_URL) -o glibc.apk
-
 clean:
        @echo -e '\e[1;31mCleaning files...\e[m'
        -rm ${OUT_ZIP}
@@ -67,4 +59,3 @@ clean:
        -rm rootfs.tar.gz
        -sudo rm -r rootfs
        -rm base.tar.gz
-       -rm glibc.apk
```

If consumer wants glibc for some specific use case, they can install it by themselves. The base WSL image would remain leaner this way.

**Comment by yuk7** [comment]
Older versions of WSL init needs glibc.
I have included glibc in the package for users of older Windows builds.

**Comment by am11** [comment]
Thanks for the reply. Perhaps in future, when the older Windows build support is not needed, a version without glibc could be released. 😊

**Comment by yuk7** [comment]
Resolved in f67217a.
Support of Windows 10 1709 by Microsoft will end in April 9, 2019 so no further support is needed.
https://support.microsoft.com/en-us/help/13853/windows-lifecycle-fact-sheet
